tsl/src/bgw_policy/retention_api.c

```diff
@@ -285,29 +287,27 @@ policy_retention_remove(PG_FUNCTION_ARGS)
 	Oid table_oid = PG_GETARG_OID(0);
 	bool if_exists = PG_GETARG_BOOL(1);
 	Cache *hcache;
+	Hypertable *hypertable;
 
 	PreventCommandIfReadOnly("remove_retention_policy()");
 
-	Hypertable *hypertable = ts_hypertable_cache_get_cache_and_entry(table_oid, true, &hcache);
+	hypertable = ts_hypertable_cache_get_cache_and_entry(table_oid, CACHE_FLAG_MISSING_OK, &hcache);
 	if (!hypertable)
 	{
-		char *view_name = get_rel_name(table_oid);
+		const char *view_name = get_rel_name(table_oid);
+
 		if (!view_name)
-		{
 			ereport(ERROR,
 					(errcode(ERRCODE_UNDEFINED_OBJECT),
-					 errmsg("OID %d does not refer to a hypertable or continuous aggregate",
-							table_oid)));
-		}
+					 errmsg("invalid hypertable or continuous aggregate")));
```

Comment:
Well... this is not really an invalid hypertable or cagg because it's neither. It's the OID that is invalid because it doesn't refer to a hypertable or cagg.

Suggest something like:
```suggestion
					 errmsg("relation is not a hypertable or continuous aggregate")));
```

or

```suggestion
					 errmsg("not a hypertable or continuous aggregate")));
```